pythran/toolchain.py

```diff
@@ -254,21 +263,47 @@ def compile_cxxfile(module_name, cxxfile, output_binary=None, **kwargs):
                           language="c++",
                           **extension_args)
 
-    try:
-        setup(name=module_name,
-              ext_modules=[extension],
-              # fake CLI call
-              script_name='setup.py',
-              script_args=['--verbose'
-                           if logger.isEnabledFor(logging.INFO)
-                           else '--quiet',
-                           'build_ext',
-                           '--build-lib', builddir,
-                           '--build-temp', buildtmp,
-                           ]
-              )
-    except SystemExit as e:
-        raise CompileError(e.args)
+    def script_runner():
+        try:
+            setup(name=module_name,
+                  ext_modules=[extension],
+                  # fake CLI call
+                  script_name='setup.py',
+                  script_args=['--verbose'
+                               if logger.isEnabledFor(logging.INFO)
+                               else '--quiet',
+                               'build_ext',
+                               '--build-lib', builddir,
+                               '--build-temp', buildtmp,
+                               ]
+                  )
+        except SystemExit as e:
+            raise CompileError(str(e))
+
```

Comment:
@pbrunet if you have time and a ready windows machine, could you try this is supported on Windows ?